src/siv.rs

```diff
@@ -137,14 +137,18 @@ where
         let iv = zero_iv_bits(&ciphertext[..IV_SIZE]);
         self.ctr.xor_in_place(&iv, &mut ciphertext[IV_SIZE..]);
 
-        let actual_tag = s2v(&mut self.mac, associated_data, &ciphertext[IV_SIZE..]);
-        if subtle::slices_equal(actual_tag.as_slice(), &ciphertext[..IV_SIZE]) != 1 {
+        let computed_tag = s2v(&mut self.mac, associated_data, &ciphertext[IV_SIZE..]);
+
+        use subtle::ConstantTimeEq;
+        let mac_is_authentic = computed_tag.as_slice().ct_eq(&ciphertext[..IV_SIZE]).into();
```

Comment:
@hdvalence you originally had a type annotation here... do you think there's a case that'd ever be necessary? `rustc` can infer the `Into` here, which implies to me it's unambiguous.

Reply:
I did it not for any particular reason other than explicitness... I have previously run into problems where there's an un-annotated type conversion, then later something changes and there's no longer unambiguous inference. This seems fine?